systemvm/debian/root/health_checks/dns_check.py

```diff
@@ -39,12 +41,15 @@ def main():
                 break
 
         if not foundEntry:
-            print "Missing entry in /etc/hosts - " + vM["ip"] + \
-                  " " + vM["vmName"]
-            exit(1)
-
-    print "All " + str(len(vMs)) + " VMs are present in /etc/hosts"
-    exit(0)
+            failedCheck = True
+            failureMessage = failureMessage + vM["ip"] + " " + vM["vmName"] + "\n"
```

Comment:
same as the dhcp check: if multiple errors we log only the last one?

Reply:
We are concatenating all errors here like previous comment.